src/jdk.jartool/share/classes/sun/tools/jar/Main.java

```diff
@@ -2318,4 +2324,18 @@ ModuleHashes computeHashes(String name) {
     static Comparator<ZipEntry> ENTRY_COMPARATOR =
         Comparator.comparing(ZipEntry::getName, ENTRYNAME_COMPARATOR);
 
+    // Set the ZipEntry dostime using date if specified otherwise the current time
+    private void setDate(ZipEntry e) {
+        setDate(e, System.currentTimeMillis());
+    }
+
+    // Set the ZipEntry dostime using the date if specified
+    // otherwise the original time
+    private void setDate(ZipEntry e, long origTime) {
+        if (date != null) {
+          e.setTimeLocal(date);
+        } else {
+          e.setTime(origTime);
+        }
```

Comment:
I think the 8 usages would be a bit clearer if renamed setZipEntryTime. Minor nit in the 2-arg setDate is that it's using 2-space indent instead of 4.

Reply:
thanks, fixed